testsupport/user_assertions.go

```diff
@@ -107,18 +108,24 @@ func VerifyResourcesProvisionedForSignup(t *testing.T, awaitilities wait.Awaitil
 	hash, err := testtier.ComputeTemplateRefsHash(tier) // we can assume the JSON marshalling will always work
 	require.NoError(t, err)
 
-	space, err := hostAwait.WaitForSpace(mur.Name,
-		wait.UntilSpaceHasTier(mur.Spec.TierName),
-		wait.UntilSpaceHasLabelWithValue(toolchainv1alpha1.SpaceCreatorLabelKey, userSignup.Name),
-		wait.UntilSpaceHasLabelWithValue(fmt.Sprintf("toolchain.dev.openshift.com/%s-tier-hash", mur.Spec.TierName), hash),
-		wait.UntilSpaceHasConditions(Provisioned()),
-		wait.UntilSpaceHasStateLabel(toolchainv1alpha1.SpaceStateLabelValueClusterAssigned),
-		wait.UntilSpaceHasStatusTargetCluster(mur.Spec.UserAccounts[0].TargetCluster))
-	require.NoError(t, err)
-
-	VerifySpaceBinding(t, hostAwait, mur.Name, space.Name, "admin")
-
-	tiers.VerifyNsTemplateSet(t, hostAwait, memberAwait, space, tierName)
+	if userSignup.Annotations[toolchainv1alpha1.SkipAutoCreateSpaceAnnotationKey] == "true" {
```

Comment:
You are testing the implementation here. Ideally our e2e test should not care about this annotation. The tests should:
- create a usersignup with the param
- check that the space is not created
The annotation is just an implementation detail.

One of the options would be to create a new function `VerifyResourcesProvisionedForSignupWithNoSpace()` or maybe add some optional params (so we don't need to update it everywhere) to `VerifyResourcesProvisionedForSignup()` to tell it if we expect the space or not.

Reply:
That's a fair point. How about this? c3f31a80066e87db1a39b6ce73ad8e911244f200

Reply:
I like this! Clean and simple.